api/s2n.h

```diff
@@ -2291,7 +2291,31 @@ S2N_API extern int s2n_shutdown_send(struct s2n_connection *conn, s2n_blocked_st
 /**
  * Used to declare what type of client certificate authentication to use.
  *
- * Currently the default for s2n-tls is for neither the server side or the client side to use Client (aka Mutual) authentication.
+ * A s2n_connection will enforce client certificate authentication (mTLS) differently based on
```

Comment:
Okay a couple of thoughts:
1. The API guide isn't really a good fit for in-depth discussions about behavior. Maybe the discussion about the default behavior is better suited to the Usage Guide, in the Client / Mutual Auth section? I dunno, you could play around with it and see if some of this information works better there.
2. You should probably take a step back if you're starting to discuss specific messages sent in the TLS handshake. Users don't care about this level of detail, they are at minimum skimming these docs and probably aren't going to read it if it goes too in-depth. The important bits are: what errors could occur with these different options, and what validation they will get with these different options. So for example:
`S2N_CERT_AUTH_OPTIONAL: Server requests a client certificate but does not error if the client doesn't send one. The certificate is validated if the client sends one.`